### PR TITLE
`bgeodb` subclient and `exact` bgeo strategy

### PR DESCRIPTION
**Pull request by menoliu**

### CODE REVIEW

**Comment by joaomcteixeira** [comment]
merged from `main` and solved the merge conflicts

File: src/idpconfgen/libs/libhigherlevel.py [hunk 1]
```diff
@@ -597,6 +602,123 @@ def validate_backbone_labels_for_torsion(labels, minimum=2):
...
 
+def get_bond_geos(fdata):
+    """
+    Calculate bond angles from structure.
+
+    Parameters
+    ----------
+    fdata : data to :pyclass:`idpconfgen.libstructure.Structure`
+
+    degrees : bool, optional
```
Comment (joaomcteixeira) [inline src/idpconfgen/libs/libhigherlevel.py:613]: these two parameters are not used in the function.

**Comment by joaomcteixeira** [comment]
* solved merge conflicts
* solved extra lint issues

i would merge #225 and add tests before merging this one.

**Comment by joaomcteixeira** [comment]
Okay. Let's go for this :+1:

**Comment by menoliu** [comment]
@joaomcteixeira `tox -e lint` returns succeeded locally, not sure why GA is saying it isn't... `tox -e test` has issues installing deps however, safe to merge?

**Comment by joaomcteixeira** [comment]
let me see

**Comment by joaomcteixeira** [comment]
let's see if tests pass now. Apart from the lint. `collections.Iterator` is not `collections.abc.Iterator` in 3.10. There was a test failing in my computer let's see here. Support for 3.10 is needed.

**Comment by menoliu** [comment]
Only looks like build timed out but only for mac_os... interesting. I plan to start adding the folded domain building option early next week and for that I need this to be merged (unless better way of branching out from this branch?)

**Comment by joaomcteixeira** [comment]
Merged indeed. sorry for the delay. Sometimes the tests to download from the PDB fail from delay

**Pull request merged**